lib/Runtime/Debug/DebugContext.h

```diff
@@ -51,11 +51,22 @@ namespace Js
         void SetHostDebugContext(HostDebugContext * hostDebugContext);
 
         void SetDebuggerMode(DebuggerMode mode);
+
         bool IsDebugContextInNonDebugMode() const { return this->debuggerMode == DebuggerMode::NotDebugging; }
         bool IsDebugContextInDebugMode() const { return this->debuggerMode == DebuggerMode::Debugging; }
         bool IsDebugContextInSourceRundownMode() const { return this->debuggerMode == DebuggerMode::SourceRundown; }
         bool IsDebugContextInSourceRundownOrDebugMode() const { return IsDebugContextInSourceRundownMode() || IsDebugContextInDebugMode(); }
 
+        bool IsDebuggerRecording() const { return this->IsDebugContextInDebugMode() && this->isDebuggerRecording; }
```

Comment:
If we are not in `Debugging` mode, such an assert would fail when we try to get the current debugging state in 
`AutoInitLibraryCodeScope` https://github.com/Microsoft/ChakraCore/pull/3447/files#diff-6858ff2e7f3b97763dd5be71dfc9ae56R1901

At current, not being in Debug mode will result in this returning false, which we then use to reset the recording state in `~AutoInitLibraryCodeScope`.

Considering that `SetIsDebuggerRecording` has no effect if we're not in `Debugging` mode, and that the Assert described above would fail, we could remove this additional condition, because it would have no effect on the state after `~AutoInitLibraryCodeScope` runs.